src/components/WorkPositions.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <div>
+    <h1>Available positions</h1>
+    <input type="checkbox" id="positionsCheckbox" v-model="positionFilter" true-value="t" false-value="f"/>
+    <label for="positionsCheckbox">Active positions only</label>
+    <ul id="positions-list">
+      <li v-for="(position, key) of filteredPositions">
+        {{ position.name }} See details
+      </li>
+    </ul>
+    <button v-if="isModerator" v-on:click="addPosition">Add position</button>
+  </div>
+</template>
+
+<script>
+  import { db } from "../App"
+  import firebase from 'firebase'
+
+  let positionsRef = db.ref('workPositions');
+  let usersRef = db.ref('users');
+
+  export default {
+    name: "work-positions",
+    firebase: {
+      positions: positionsRef,
+      users: usersRef
+    },
+    data () {
+      return {
+        positionFilter: ""
+      }
+    },
+    methods: {
+      addPosition: function () {
+        window.alert('Not implemented yet')
+      }
+    },
+    computed: {
+      filteredPositions: function () {
+        if (this.positionFilter === 't') {
+          return this.positions.filter((position) => {
+            return position.isActive;
+          })
+        }
+        return this.positions
+      },
+      isModerator: function () {
```

Comment:
Czy zamiast pobierać wszystkich użytkowników i ich filtrować, nie lepiej zrobić tylko czegoś typu `db.ref('users').ref(uid)` ?

Reply:
Nie, ponieważ:
żeby mieć uid muszę wywołać `firebase.auth().currentUser;`
jeśli wywołam to przed `export default` to zwróci mi nulla
a muszę wywołać przed, bo ten uid muszę wrzucić do obiektu `firebase`

Reply:
A dlaczego nie możesz wewnątrz metody zrobić najpierw `firebase.auth().currentUser` a potem dopiero odwolac sie do bazy?

Reply:
No bo do bazy odwołuję się wewnątrz obiektu firebase, do której musze z tego co rozumiem od razu wrzucić odpowiednie refy

Reply:
Poza tym jakbym jakoś próbował się wewnątrz metody odwoływać do bazy z pominięciem obiektu `firebase` to pewnie musiałoby to iść przez wywołanie asynchroniczne, co jest słabym pomysłem w metodzie która musi zwrócić boola xd

Reply:
no coz, ja bym to zrobil inaczej ale juz nie bede sie czepial, dziala to dam spokoj ;)